contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,334 @@
+
+    constructor(address signer_) ECDSASigner(signer_) {}
+
+    function ongoingAuction(uint256 id_) external view returns (bool) {
+        // check the auction status and return
+        require(
+            auctions[id_].id > 0,
+            "FeralfileEnglishAuction: auction not found"
+        );
+
+        return
+            auctions[id_].startAt <= block.timestamp &&
+            auctions[id_].endAt >= block.timestamp;
+    }
+
+    function registerAuctions(Auction[] calldata auctions_) external onlyOwner {
+        // for-loop to set the auction
+        // able to reset an auction if there's no winner and highest bid = 0
+        for (uint256 i = 0; i < auctions_.length; i++) {
+            Auction memory auc_ = auctions[auctions_[i].id];
+
+            require(
+                auc_.id == 0,
+                "FeralfileEnglishAuction: auction already exist"
+            );
+
+            require(
+                auctions_[i].id > 0,
+                "FeralfileEnglishAuction: invalid auction id"
+            );
+
+            require(
+                auctions_[i].startAt >= block.timestamp &&
+                    auctions_[i].endAt > block.timestamp,
+                "FeralfileEnglishAuction: auction start time and end time should be in the future"
+            );
+
+            require(
+                auctions_[i].endAt > auctions_[i].startAt,
+                "FeralfileEnglishAuction: auction end time should be after start time"
+            );
+
+            require(
+                auctions_[i].minIncreaseFactor > 0,
+                "FeralfileEnglishAuction: auction min increase factor should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseAmount > 0,
+                "FeralfileEnglishAuction: auction min increase amount should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minPrice > 0,
+                "FeralfileEnglishAuction: auction min price should be greater than 0"
+            );
+
+            auctions[auctions_[i].id] = auctions_[i];
+        }
+    }
+
+    function listHighestBids(
+        uint256[] memory aucIds_
+    ) external view returns (LatestAuctionStatus[] memory) {
+        LatestAuctionStatus[] memory results = new LatestAuctionStatus[](
+            aucIds_.length
+        );
+        for (uint i = 0; i < aucIds_.length; i++) {
+            // get the auction object
+            Auction memory auction_ = auctions[aucIds_[i]];
+            // get the last auction bid object
+            results[i].latestBid = highestBids[aucIds_[i]];
+            results[i].endAt = auction_.endAt;
+            results[i].isSettled = auction_.isSettled;
+        }
+        return results;
+    }
+
+    function isValidBidRequest(
+        uint256 auctionID_,
+        address bidder_,
+        uint256 amount_
+    ) external view returns (bool) {
+        // get the auction object
```

Comment:
I see you left some the same comments in different functions but to me, it could be redundant. Those are pretty easy to understand and the code can explain itself.